app/controllers/api/base_controller/generic.rb

```diff
@@ -94,6 +94,23 @@ def delete_resource(type, id = nil, _data = nil)
         delete_resource_action(klass, type, id)
       end
 
+      def copy_resource(type, id, data = {})
+        klass    = collection_class(type)
+        resource = resource_search(id, type, klass)
+        attributes = data.dup
+        new_resource = resource.dup
+        attributes.each do |attr, value|
+          if new_resource.respond_to? "#{attr}="
+            new_resource.send("#{attr}=", value)
+          else
+            raise ActiveModel::UnknownAttributeError.new(self, attr)
```

Comment:
would not raise an `UnknownAttributeError` since that'll be handled by the rescue below. Would just do a `raise "uknown attribute #{attr}"